chain/rpc.go

```diff
@@ -163,6 +166,96 @@ func (c *RPCClient) BlockStamp() (*waddrmgr.BlockStamp, error) {
 	}
 }
 
+// FilterBlocks scans the blocks contained in the FilterBlocksRequest for any
+// addresses of interest. For each requested block, the corresponding compact
+// filter will first be checked for matches, skipping those that do not report
+// anything. If the filter returns a postive match, the full block will be
+// fetched and filtered. This method returns a FilterBlocksReponse for the first
+// block containing a matching address. If no matches are found in the range of
+// blocks requested, the returned response will be nil.
+func (c *RPCClient) FilterBlocks(
+	req *FilterBlocksRequest) (*FilterBlocksResponse, error) {
+
+	blockFilterer := NewBlockFilterer(
+		c.chainParams, req.ExternalAddrs, req.InternalAddrs,
+	)
+
+	// Construct a watch list containing the script addresses of all
+	// internal and external addresses that were requested.
+	nAddrs := len(req.ExternalAddrs) + len(req.InternalAddrs)
+	watchList := make([][]byte, 0, nAddrs)
+	for _, addr := range req.ExternalAddrs {
+		watchList = append(watchList, addr.ScriptAddress())
+	}
+	for _, addr := range req.InternalAddrs {
+		watchList = append(watchList, addr.ScriptAddress())
+	}
+
+	// Iterate over the requested blocks, fetching the compact filter for
+	// each one, and matching it against the watchlist generated above. If
+	// the filter returns a positive match, the full block is then requested
+	// and scanned for addresses using the block filterer.
+	for i, blk := range req.Blocks {
+		rawFilter, err := c.GetCFilter(&blk.Hash, wire.GCSFilterRegular)
```

Comment:
I think the RPC (would need to check tho) has actually been updated in master. As a result, we can actually fetch the _entire_ filter range in a single command. Not a blocker though, and we can simply add a TOOD (with a corresponding issue) instead.